Doc/c-api/init.rst

```diff
@@ -871,6 +871,17 @@ code, or when embedding the Python interpreter:
    the caller needn't check for ``NULL``).
 
 
+.. c:function:: PyThreadState* PyThreadState_GetUnsafe()
+
+   Similar to :c:func:`PyThreadState_Get`, but don't kill the process with a
+   fatal error if it is NULL. The caller is responsible to check if the result
+   is NULL.
```

Comment:
IMO, the fact that this returns NULL without setting an exception should be explicitly mentioned:
```suggestion
   is NULL.
   This function does not (and cannot) set an exception when it returns NULL.
```

That the question: is there any future where this *might* need to raise an exception? Probably not, since if this fails you can't really use any other API.

Reply:
If the caller doesn't hold the GIL, the function returns NULL. In this case, it cannot set an exception, since the GIL must be held to call the C API, like function to raise an exception.

IMO it's ok to leave it like that in the doc. If the API changes, the change can be documented here.